classes/helper/HelperList.php

```diff
@@ -826,4 +818,64 @@ public function displayListFooter()
 
         return $this->footer_tpl->fetch();
     }
+
+    /**
+     * @param string $token
+     * @param int $id
+     *
+     * @return string
+     *
+     * @throws PrestaShopException
+     */
+    protected function getViewLink($token, $id)
+    {
+        switch ($this->table) {
+            case 'customer':
```

Comment:
Ok clearly it's a quick fix, the thing is do I extend this behavior to other migrated entities as well?

Reply:
😅
Yeah, but this violates the O of SOLID.

Reply:
do you really want me to change this? ^^
I could try and perform a generic change but I'm afraid of the side effects..

Reply:
Since the issue is a trivial one, I'm not sure the cost is worth it :s

Reply:
sadly I agree.. it's just that we will have to add other switch cases as we continue migrating pages
by the way do you think there are other pages that would need to be managed? based on the contents managed in the quick search:
product, customer, order, invoices, cart, modules
There might be a bug on the module links, but I don't know what was the expected result?

Reply:
The more we dig, the worse it gets 😄

Reply:
This is hard to fix for several reasons:

- If we use getAdminLink() then it's a breaking change (in this class' contract, links are meant to be built following a specific schema)
- If we don't, then the component will be more and more broken as we migrate pages
- We cannot inject dependencies because it would be a BC

Basically, it's broken by design, but we cannot let it stay broken.

My suggestion:

1. Extract these two methods into an interface.
2. Create at least two implementations, one for the previous link generation, one for the new one.
3. Use a factory to build the correct implementation according to parameters.
4. It needs to be covered by tests
3. It must support all the migrated CRUD pages.